amical/mf_pipeline/ami_function.py

```diff
@@ -400,6 +404,8 @@ def make_mf(
 
     if display:
         _plot_mask_coord(xy_coords, maskname, instrument)
+        if save:
+            plt.savefig(figdir + Path(filename).stem + "_1.pdf")
```

Comment:
This can't work if figdir is left to it's default value (`None`)
I suggest refactoring this so that we have a single argument `save_to` (or `save_dir`, up to you) that replaces `figdir` and `save`